src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -340,22 +349,55 @@ public Activity(string operationName)
 
         /// <summary>
         /// Update the Activity to have a tag with an additional 'key' and value 'value'.
-        /// This shows up in the <see cref="Tags"/>  enumeration.   It is meant for information that
+        /// This shows up in the <see cref="Tags"/>  enumeration. It is meant for information that
         /// is useful to log but not needed for runtime control (for the latter, <see cref="Baggage"/>)
         /// </summary>
         /// <returns>'this' for convenient chaining</returns>
-        public Activity AddTag(string key, string? value)
+        /// <param name="key">The tag key name</param>
+        /// <param name="value">The tag value mapped to the input key</param>
+        public Activity AddTag(string key, string? value) => AddTag(key, (object?) value);
+
+        /// <summary>
+        /// Update the Activity to have a tag with an additional 'key' and value 'value'.
+        /// This shows up in the <see cref="TagObjects"/> enumeration. It is meant for information that
+        /// is useful to log but not needed for runtime control (for the latter, <see cref="Baggage"/>)
+        /// </summary>
+        /// <returns>'this' for convenient chaining</returns>
+        /// <param name="key">The tag key name</param>
+        /// <param name="value">The tag value mapped to the input key</param>
+        public Activity AddTag(string key, object? value)
         {
-            KeyValuePair<string, string?> kvp = new KeyValuePair<string, string?>(key, value);
+            KeyValuePair<string, object?> kvp = new KeyValuePair<string, object?>(key, value);
 
-            if (_tags != null || Interlocked.CompareExchange(ref _tags, new LinkedList<KeyValuePair<string, string?>>(kvp), null) != null)
+            if (_tags != null || Interlocked.CompareExchange(ref _tags, new TagsLinkedList(kvp), null) != null)
             {
                 _tags.Add(kvp);
             }
 
             return this;
         }
 
+        /// <summary>
+        /// Add or update the Activity tags with the input key and value.
+        /// If there is any tag added before with the same key, then the value mapped to this key will be updated with the new input value.
+        /// If the input value is null and key exist before, this entry with that key will be removed from the collection. otherwise nothing wil happen.
```

Comment:
nit: Looks like some of the summary text needs to be cleaned up: `collection. otherwise nothing wil happen. otherwise, the key and value will get added to the attributes list.` Huh? 🤣

Reply:
I'll try to cleanup that.

Reply:
This is done!